### PR TITLE
Add support for message with content type and user properties

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Add parameters to SendMessage.
- Add overloaded SendMessage method.
- Add Unit Tests for DeviceClient.
- Enable running Unit Test on pipeline.

## Motivation and Context
- Resolves nanoFramework/Home#1171.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [x] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [x] I have added new tests to cover my changes.

### CODE REVIEW

File: nanoFramework.Azure.Devices.Client/DeviceClient.cs [hunk 1]
```diff
@@ -382,25 +387,67 @@ public void RemoveMethodCallback(MethodCallback methodCallback)
...
+            string contentType,
+            ArrayList userProperties,
+            CancellationToken cancellationToken = default,
+            string dtdlComponentname = "")
         {
-            string topic = _telemetryTopic;
+            StringBuilder topic = new(_telemetryTopic);
+
+            // add content type to property bag, if there is one
+            if(!string.IsNullOrEmpty(contentType))
```
Comment (Ellerbach) [inline nanoFramework.Azure.Devices.Client/DeviceClient.cs:425]: ctrl + k + d should format properly the file

File: nanoFramework.Azure.Devices.Client/DeviceClient.cs [hunk 2]
```diff
@@ -382,25 +387,67 @@ public void RemoveMethodCallback(MethodCallback methodCallback)
...
+            {
+                topic.Append(EncodeContentType(contentType));
+                topic.Append("&");
+            }
+
+            // add user properties to property bag, if they exist
+            if(userProperties.Count > 0)
```
Comment (Ellerbach) [inline nanoFramework.Azure.Devices.Client/DeviceClient.cs:432]: please check for null before checking for the count

File: nanoFramework.Azure.Devices.Client/DeviceClient.cs [hunk 3]
```diff
@@ -382,25 +387,67 @@ public void RemoveMethodCallback(MethodCallback methodCallback)
...
+            {
+                topic.Append(EncodeUserProperties(userProperties));
+                topic.Append("&");
+            }
 
             if (!string.IsNullOrEmpty(dtdlComponentname))
             {
                 topic += $"$.sub={HttpUtility.UrlEncode(dtdlComponentname)}";
```
Comment (Ellerbach) [inline nanoFramework.Azure.Devices.Client/DeviceClient.cs:440]: most likely adjust this for the Append as well pattern

File: nanoFramework.Azure.Devices.Client/DeviceClient.cs [hunk 4]
```diff
@@ -382,25 +387,67 @@ public void RemoveMethodCallback(MethodCallback methodCallback)
...
             }
```
Comment (Ellerbach) [inline nanoFramework.Azure.Devices.Client/DeviceClient.cs:442]: remove the last & if there is one

**Review by Ellerbach — COMMENTED** [review]
Few elements to adjust but looks already good

File: nanoFramework.Azure.Devices.Client/DeviceClient.cs [hunk 5]
```diff
@@ -382,25 +387,75 @@ public void RemoveMethodCallback(MethodCallback methodCallback)
         /// Send a message to Azure IoT.
         /// </summary>
         /// <param name="message">The message to send.</param>
+        /// <param name="contentType">Content of the application message.</param>
+        /// <param name="cancellationToken">A cancellation token. If you use the default one, the confirmation of delivery will not be awaited.</param>
+        /// <returns>True for successful message delivery.</returns>
+        public bool SendMessage(
```
Comment (Ellerbach) [inline nanoFramework.Azure.Devices.Client/DeviceClient.cs:393]: just to make sure: we are not breaking any compatibility with the changes made here? If yes, then please put back the existing function as well

File: nanoFramework.Azure.Devices.Client/DeviceClient.cs [hunk 6]
```diff
@@ -382,25 +387,75 @@ public void RemoveMethodCallback(MethodCallback methodCallback)
...
+
+            // add content type to property bag, if there is one
+            if (!string.IsNullOrEmpty(contentType))
+            {
+                topic.Append(EncodeContentType(contentType));
+                topic.Append("&");
+            }
+
+            // add user properties to property bag, if they exist
+            if (userProperties.Count > 0)
```
Comment (Ellerbach) [inline nanoFramework.Azure.Devices.Client/DeviceClient.cs:432]: user properties can be null, so would do:
```suggestion
            if (userProperties != null) && (userProperties.Count > 0))
```

File: nanoFramework.Azure.Devices.Client/DeviceClient.cs (hunk 5 above)
Comment (josesimoes) [reply nanoFramework.Azure.Devices.Client/DeviceClient.cs:393]: OK I've added it.

**Review by Ellerbach — APPROVED** [review]
Now looks good to me :-) Thanks a lot!

**Pull request merged**